src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,1301 @@
+		private const string SR_TabViewScrollDecreaseButtonTooltip = "TabViewScrollDecreaseButtonTooltip";
+		private const string SR_TabViewScrollIncreaseButtonTooltip = "TabViewScrollIncreaseButtonTooltip";
+
+		private readonly DispatcherHelper m_dispatcherHelper;
+
+		private ContentPresenter m_tabContentPresenter = null;
+		private ContentPresenter m_rightContentPresenter = null;
+		private ColumnDefinition m_leftContentColumn;
+		private ColumnDefinition m_tabColumn;
+		private ColumnDefinition m_addButtonColumn;
+		private ColumnDefinition m_rightContentColumn;
+		private Button m_addButton;
+		private RepeatButton m_scrollDecreaseButton;
+		private RepeatButton m_scrollIncreaseButton;
+		private ScrollViewer m_scrollViewer;
+		private ItemsPresenter m_itemsPresenter;
+		private Grid m_shadowReceiver;
+		private Grid m_tabContainerGrid;
+		private ListView m_listView;
+		private Size previousAvailableSize;
+
+		private bool m_updateTabWidthOnPointerLeave = false;
+		private string m_tabCloseButtonTooltipText;
+
+		private long m_listViewCanReorderItemsPropertyChangedRevoker;
+		private long m_listViewAllowDropPropertyChangedRevoker;
+
+		public TabView()
+		{
+			m_dispatcherHelper = new DispatcherHelper(this);
+
+			var items = new ObservableVector<object>();
+			SetValue(TabItemsProperty, items);
+
+			SetDefaultStyleKey(this);
+
+			this.Loaded += OnLoaded;
+
+			// KeyboardAccelerator is only available on RS3+
+			if (SharedHelpers.IsRS3OrHigher())
+			{
+				KeyboardAccelerator ctrlf4Accel = new KeyboardAccelerator();
+				ctrlf4Accel.Key = VirtualKey.F4;
+				ctrlf4Accel.Modifiers = VirtualKeyModifiers.Control;
+				ctrlf4Accel.Invoked += OnCtrlF4Invoked;
+				ctrlf4Accel.ScopeOwner = this;
+				KeyboardAccelerators.Add(ctrlf4Accel);
+
+				m_tabCloseButtonTooltipText = ResourceAccessor.GetLocalizedStringResource(SR_TabViewCloseButtonTooltipWithKA);
+			}
+			else
+			{
+				m_tabCloseButtonTooltipText = ResourceAccessor.GetLocalizedStringResource(SR_TabViewCloseButtonTooltip);
+			}
+
+			// Ctrl+Tab as a KeyboardAccelerator only works on 19H1+
+			if (SharedHelpers.Is19H1OrHigher())
+			{
+				KeyboardAccelerator ctrlTabAccel = new KeyboardAccelerator();
+				ctrlTabAccel.Key = VirtualKey.Tab;
+				ctrlTabAccel.Modifiers = VirtualKeyModifiers.Control;
+				ctrlTabAccel.Invoked += OnCtrlTabInvoked;
+				ctrlTabAccel.ScopeOwner = this;
+				KeyboardAccelerators.Add(ctrlTabAccel);
+
+				KeyboardAccelerator ctrlShiftTabAccel = new KeyboardAccelerator();
+				ctrlShiftTabAccel.Key = VirtualKey.Tab;
+				ctrlShiftTabAccel.Modifiers = VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift;
+				ctrlShiftTabAccel.Invoked += OnCtrlShiftTabInvoked;
+				ctrlShiftTabAccel.ScopeOwner = this;
+				KeyboardAccelerators.Add(ctrlShiftTabAccel);
+			}
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+
+			UnhookEventsAndClearFields();
+
+			//IControlProtected controlProtected{ *this };
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)